sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/package.scala

```diff
@@ -88,7 +88,9 @@ package object expressions  {
   /**
    * A helper function to bind given expressions to an input schema.
    */
-  def toBoundExprs(exprs: Seq[Expression], inputSchema: Seq[Attribute]): Seq[Expression] = {
+  def toBoundExprs[A <: Expression](
```

Comment:
I would like to minimize the chance that future changes suffer from the same issue. In order to do that we should provide API in a logical place, it does not make a whole lot of sense to me that I need to look in `package.scala` to find a more performant version of `BindReferences.bindReference(..)` for a seq. Can we move this function to `BindReference` and name it `bindReferences`?